src/Compilers/Server/VBCSCompilerTests/DesktopBuildClientTests.cs

```diff
@@ -22,23 +22,32 @@ public abstract class DesktopBuildClientTests : TestBase
     {
         private sealed class TestableDesktopBuildClient : DesktopBuildClient
         {
+            /// <summary>
+            /// Stores fully constructed pipe name.
+            /// </summary>
             private readonly string _pipeName;
+            /// <summary>
+            /// Stores unhashed "uniqueifying" pipe name component.
```

Comment:
I think this makes the code a little too complex, though. The primary reason we provide this switch is to take the safeties off and let users define their own semantics. By making it a component we reduce predictability and potentially just make things more difficult. For example, if you made your own client to talk to the server, now you still have to reverse engineer the other parts of the naming.

I think we should refrain from trying to be too clever here.